sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveCommandSuite.scala

```diff
@@ -290,6 +291,29 @@ class HiveCommandSuite extends QueryTest with SQLTestUtils with TestHiveSingleto
       checkAnswer(
         sql("SELECT employeeID, employeeName FROM part_table WHERE c = '2' AND d = '1'"),
         sql("SELECT * FROM non_part_table").collect())
+
+      // check for case insensitive property of partition column name in load command.
+      withSQLConf(SQLConf.CASE_SENSITIVE.key -> "false") {
+        withInputFile { f =>
+          sql(s"""$loadQuery INPATH "${f.toURI}" INTO TABLE part_table PARTITION(C="1", D="2")""")
+        }
+        checkAnswer(
+          sql("SELECT employeeID, employeeName FROM part_table WHERE c = '2' AND d = '1'"),
+          sql("SELECT * FROM non_part_table").collect())
+      }
+    }
+  }
+
+  test("case insensitive property of partition column name in insert command") {
+    // check for case insensitive property of partition column name in insert command.
+    withTable("part_table") {
+      withSQLConf(SQLConf.CASE_SENSITIVE.key -> "false") {
+        sql("CREATE TABLE part_table (price int, qty int) partitioned by (year int, month int)")
+        sql("INSERT INTO part_table PARTITION(YEar = 2015, month = 1) SELECT 1, 1")
```

Comment:
BTW, this test case only covers `Static` Partition.
We need dynamic partitioning test case, too.

Reply:
Added, please let me the know the added test-case is sufficient or not, thanks for your valuable suggestions and time.